### ISSUE TITLE
Related Getters Returns Old Data After Updating Resource

### ISSUE THREAD

**Issue by yoelpc4** [reststate/reststate-vuex#161]
Hi @CodingItWrong thanks for helpful package.

Currently I'm integrating restate-vuex with vuetify v-data-table to make a stateful data table. 

Here is my data table component
https://gist.github.com/yoelpc4/beed885b72b1082c27208ce64ded98e9

My edit province component
https://gist.github.com/yoelpc4/58093a0d447064e0cf1aca0b80be1d08

And this is my province view
https://gist.github.com/yoelpc4/cf88bcdfe88ff2c60397ae5e1a12e7c4

I binded v-data-table items with items data-table computed property so that, whenever `provinces/update` action is dispatched from edit province component, the computed items in data-table component is invoked. 

The `provinces/all` getter is working as expected returning province resource with updated name. Problem arised when `countries/related` getter is invoked, because it keeps returns the updated province with old country instead of new country. 

After console logging `restate-vuex.js`, I found that `line 371` is the root of my problem.

```js
const related = state.related.find(matches(relationshipIndex));
```

Province records before update
https://gist.github.com/yoelpc4/4d94605e981ce834a2bc059645b7036b

Province records after update
https://gist.github.com/yoelpc4/f58322e821f150748e816d01881b295b

Notice the first record before & after update in store is updated, but `countries/related` getters always returns the old related data instead of the new one.

Maybe you can reproduce my issue in advance to resolve this issue.

**Comment by CodingItWrong** [comment]
Hi, I think I follow the problem you're describing. Let me ask a few followup questions to make sure I understand in detail, and then I can investigate.

- Can you send me your Vuex store config so I can be sure to understand how you're calling reststate-vuex?
- "The provinces/all getter is working as expected returning province resource with updated name. Problem arised when countries/related getter is invoked, because it keeps returns the updated province with old country instead of new country." Can you describe the situation in more detail? Are you updating the country that a province is connected to? And then when you request the related countries for that province, you are getting the old country instead of the new country?

**Comment by yoelpc4** [comment]
Thanks for your attention.

- My vuex store config
```js
import Vue from 'vue'
import Vuex from 'vuex'
import { mapResourceModules } from '@reststate/vuex'
import config from '@/config/app'
import jsonApi from '@/modules/jsonApi/index'

Vue.use(Vuex)

export default new Vuex.Store({
  strict: config.env !== 'production',
  modules: {
    ...mapResourceModules({
      names: [
        'areas',
        'attachments',
        'cities',
        'continents',
        'countries',
        'currencies',
        'permissions',
        'projects',
        'provinces',
        'roles',
        'strategic-business-units',
        'countries',
        'users',
        'work-unit-achievements',
        'work-units',
        'work-types'
      ],
      httpClient: jsonApi.client
    })
  }
})
```

Just in case you're curious about my axios client instance
```js
import axios from 'axios'
import nprogress from 'nprogress'
import auth from '@/modules/auth/index'
import config from '@/modules/jsonApi/config'
import locale from '@/modules/locale/index'

const loadToken = config => {
  if (auth.check()) {
    // if token needs to be refreshed, refresh it
    if (auth.token.needRefresh()) {
      auth.refresh()
    }

    // set authorization header with bearer token
    config.headers.common['Authorization'] = `Bearer ${auth.token.value}`
  } else {
    auth.handleUnauthenticated()
  }

  return config
}

const loadLocale = config => {
  if (locale.lang.code) {
    // set locale header with lang
    config.headers.common['Locale'] = locale.lang.code
  }

  return config
}

const fulfillRequest = config => {
  // start progress bar
  nprogress.start()

  // load token before dispatching request to server
  loadToken(config)

  // load locale before dispatching request to server
  loadLocale(config)

  return config
}

const rejectRequest = error => {
  // finish progress bar
  nprogress.done()

  // handle on unauthenticated
  if (error.response.status === 401) {
    auth.handleUnauthenticated()
  }

  return Promise.reject(error)
}

const fulfillResponse = response => {
  // finish progress bar
  nprogress.done()

  return response
}

const rejectResponse = error => {
  // finish progress bar
  nprogress.done()

  // handle on unauthenticated
  if (error.response.status === 401) {
    auth.handleUnauthenticated()
  }

  return Promise.reject(error)
}

const client = axios.create({
  baseURL: config.baseUrl,
  headers: config.headers
})

client.interceptors.request.use(fulfillRequest, rejectRequest)

client.interceptors.response.use(fulfillResponse, rejectResponse)

export default client
```

- Yes I updating the related country of province. This is an example for your hunch
Record before update
```js
const province = {
  type: 'provinces',
  id: '1',
  attributes: {
    name: 'Georgia'
  },
  relationships: {
    country: {
      data: {
        type: 'countries',
        id: '1'
      }
    }
  },
  included: [
    {
      type: 'countries',
      id: '1',
      attributes: {
        name: 'United States'
      }
    }
  ]
}
```
Expected record after update
```js
const province = {
  type: 'provinces',
  id: '1',
  attributes: {
    name: 'Georgia Updated'
  },
  relationships: {
    country: {
      data: {
        type: 'countries',
        id: '2'
      }
    }
  },
  included: [
    {
      type: 'countries',
      id: '2',
      attributes: {
        name: 'Canada'
      }
    }
  ]
}
```
Just after the update action invoked, I retrieves all data with `const data = this.$store.getters['provinces/all'] || []`. The first province record has Georgia Updated as name and this is perfectly works. After that `data` array is mapped to formatting the province records with 
```js
onFormatItems(data) {
  return data
    .map(province => {
      const options = {
          parent: {
            type: 'provinces',
            id: province.id
          },
          relationship: 'country'
        }

        const related = this.$store.getters['countries/related'](options)

        province['country'] = related

      return province
    })
},
```
Instead of Canada as related country of the first province, it has United States as related country.

**Comment by CodingItWrong** [comment]
@yoelpc4 Sorry for my long delay getting back to you; I didn't see the email notification about your reply.

This has now been fixed in #184 and released as part of `0.2.1`. Please give it a try and let me know if it fixes the issue for you. Thanks for reporting this.
